plugin.video.otaku/resources/lib/pages/nyaa.py

```diff
@@ -169,9 +169,8 @@ def get_sources(self, query, anilist_id, episode, status, media_type, rescrape):
         sources = self._get_episode_sources(query, anilist_id, episode, status, rescrape)
 
         if not sources and ':' in query:
-            q1, q2 = query.split('|')
-            q1 = q1[1:-1].split(':')[0]
-            q2 = q2[1:-1].split(':')[0]
+            pattern = r'\|?.+?:([^|]+)'
```

Comment:
Is this right? earlier we were dropping the text after the `:`,  but with this you are dropping the text before the `:` which is usually the Anime title. So the query may get completely unrelated anime results due to this. please check

Reply:
Ahh yes you're right, I went over it again and believe I've recommited changes, not sure if it's 100 proper as I'm away from home right now, only reason I edited the q1 and q2 is on the previous torrent update that was reverted it would stall nyaa on certain stuff and not retrieve anything. Thank you for looking btw

Reply:
q1, q2 = (q.split(':')[0] for q in query[1:-1].split('|'))
    query2 = f"({q1})|({q2})"
ok so, This modification aligns the logic of the committed code snippet more closely with the original if this line works better. or let me know if i messed this up haha thank you